test/compile-test.cc

```diff
@@ -301,77 +309,164 @@ template <size_t max_string_length, typename Char = char> struct test_string {
 
 template <size_t max_string_length, typename Char = char, typename... Args>
 consteval auto test_format(auto format, const Args&... args) {
-  test_string<max_string_length, Char> string{};
+  fixed_length_string<max_string_length, Char> string{};
   fmt::format_to(string.buffer, format, args...);
   return string;
 }
+#endif
+
+#if defined(__cpp_constexpr) && (__cpp_constexpr >= 202211L) && \
+    defined(FMT_CONSTEXPR20) && FMT_USE_NONTYPE_TEMPLATE_ARGS
```

Comment:
``FMT_CONSTEXPR20`` is always defined:

https://github.com/fmtlib/fmt/blob/fb07b37c5b104e98a9316668c8dbbed166c9b873/include/fmt/base.h#L133-L139

Reply:
Will remove it, the other checks guarantee C++23 anyway.